### DISCUSSION TITLE
CCbutton, Ignore note off?

### DISCUSSION THREAD

**Discussion by Vonholt** [tttapa/Control-Surface#942] [Q&A] [answered]
Hello, 

Title basically says it all but I am also trying to figure out any solution. I am trying to implement two buttons, Button 1 (pin2) sends CC29 with a value of 1, button 2 (pin3) sends CC29 with a value of 127, 

CCButton gets me halfway there but I need to ignore the 0 value when the buttons are released.

`CCButton button[] {
{2, 29, 1}, //Mode Down
 {3, 29, 127} //Mode Up` 

![image](https://github.com/tttapa/Control-Surface/assets/7227319/031fabe3-f54a-416b-b64f-3bf151f58bdc)

I can make the release value 127 but that fires the message twice on button 2.

`CCButton button[] {
{2, 29, 1}, //Mode Down
 {3, 29, {127, 127}} //Mode Up` 

![image](https://github.com/tttapa/Control-Surface/assets/7227319/a401b306-472b-4b88-a4a4-5919e4035756)

I also tried using CCIncrementDecrementButtons. It works fine for button 1 sending the 1 value but I cannot figure out how to send 127 on button 2.

Any help would be greatly appreciated.

**Comment by tttapa** [comment, marked as answer]
If you want to customize the behavior of the provided classes, I recommend studying the techniques covered in https://tttapa.github.io/Control-Surface-doc/Doxygen/d4/d3b/Custom-MIDI-Output-Element_8ino-example.html.

**Reply by Vonholt** [reply]
Thank you! Fiddled with it for a couple hours but got the class set up how I wanted. Copied it over to the main code and now the whole controller is working as I need it to. As a very novice hobby coder this is a great library. Not too simple but just complicated enough to keep you learning. Thanks!
